web/src/user_group_edit.js

```diff
@@ -271,15 +271,41 @@ function update_group_membership_button(group_id) {
     }
 }
 
-export function handle_subgroup_edit_event(group_id) {
+export function handle_subgroup_edit_event(group_id, subgroup_ids) {
     if (!overlays.groups_open()) {
         return;
     }
-    const group = user_groups.get_user_group_from_id(group_id);
+
+    const active_group_id = get_active_data().id;
+    if (active_group_id === undefined) {
+        return;
+    }
+
+    const current_user_id = people.my_current_user_id();
 
     // update members list if currently rendered.
-    if (is_editing_group(group_id)) {
+    if (group_id === active_group_id) {
+        const group = user_groups.get_user_group_from_id(group_id);
         user_group_edit_members.update_member_list_widget(group);
+
+        if (
+            !user_groups.is_direct_member_of(current_user_id, group_id) &&
+            user_groups.is_user_in_any_group(subgroup_ids, current_user_id)
```

Comment:
Does this check need to be recursive?

Reply:
I guess `is_user_in_any_group` does that. Might be nice for the function to accept an explicit `direct_member_only` parameter with a default of false, like `is_user_in_group`, just to make it easier to known its behavior when viewing its definition.

Reply:
Updated `is_user_in_any_group` to accept `direct_member_only` parameter.